onnxruntime/python/tools/transformers/dynamo_onnx_helper.py

```diff
@@ -0,0 +1,92 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+
+import onnx
+
+
+class DynamoOnnxHelper:
+    """
+    Helper class for processing ONNX models exported by torch Dynamo.
+    """
+
+    def __init__(self, model: onnx.ModelProto):
+        self.model = model
+
+    def update_edges(self, edge_mapping: dict) -> None:
+        """
+        Updates the edges in the model according to the given mapping.
+        """
+        for node in self.model.graph.node:
+            for i in range(len(node.input)):
+                if node.input[i] in edge_mapping:
+                    node.input[i] = edge_mapping[node.input[i]]
+            for i in range(len(node.output)):
+                if node.output[i] in edge_mapping:
+                    node.output[i] = edge_mapping[node.output[i]]
+
+        for graph_input in self.model.graph.input:
+            if graph_input.name in edge_mapping:
+                graph_input.name = edge_mapping[graph_input.name]
+        for graph_output in self.model.graph.output:
+            if graph_output.name in edge_mapping:
+                graph_output.name = edge_mapping[graph_output.name]
+
+    def unroll_function(self, func_name: str) -> None:
+        """
+        Unrolls the function with the given name in the model.
```

Comment:
Can this be done with the onnx inliner? https://onnx.ai/onnx/api/inliner.html#inline-selected-functions

Reply:
unfortunately no

Reply:
Was there any limitations? Could you share more? Thanks!

Reply:
first, the function unrolling is semantically different from inliner. the former does not inline the function recursively. second, I played with inliner initially, but the output onnx file was invalid. you can take a try with the original model exported from dynamo

Reply:
cc @gramalingam @BowenBao

Reply:
Re. the first point: I guess you mean you want to do selective inlining (may be just a single function)? (The inliner API does support that.) Re. the second point: where can I find the original model? Thanks!

Reply:
onnx inliner has an argument `function_ids` to specify which function to inline by function (domain, name). For the invalid output onnx issue, did you try with latest onnx release? @wangyems

Reply:
@BowenBao I tried the function_ids at the very beginning a few weeks ago but let me try with the main branch 
@gramalingam you can get the original file by running convert_to_onnx.py without argument https://github.com/microsoft/onnxruntime/tree/wangye/phi2_doc/onnxruntime/python/tools/transformers/models/phi2#readme.